src/Persistence/Proxy.php

```diff
@@ -0,0 +1,396 @@
+<?php
+
+/*
+ * This file is part of the zenstruck/foundry package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Foundry\Persistence;
+
+use Doctrine\ODM\MongoDB\DocumentManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ObjectManager;
+use Zenstruck\Assert;
+use Zenstruck\Callback;
+use Zenstruck\Callback\Parameter;
+use Zenstruck\Foundry\Factory;
+use Zenstruck\Foundry\Instantiator;
+use Zenstruck\Foundry\RepositoryProxy;
+
+/**
+ * @template TProxiedObject of object
+ * @mixin TProxiedObject
+ *
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @final
+ */
+class Proxy implements \Stringable
+{
+    /**
+     * @phpstan-var class-string<TProxiedObject>
+     */
+    private string $class;
+
+    private bool $autoRefresh;
+
+    private bool $persisted = false;
+
+    /**
+     * @internal
+     *
+     * @phpstan-param TProxiedObject $object
+     */
+    public function __construct(
+        /** @param TProxiedObject $object */
+        private object $object
+    ) {
+        $this->class = $object::class;
+        $this->autoRefresh = Factory::configuration()->defaultProxyAutoRefresh();
+    }
+
+    public function __call(string $method, array $arguments) // @phpstan-ignore-line
+    {
+        return $this->_real()->{$method}(...$arguments);
+    }
+
+    public function __get(string $name): mixed
+    {
+        return $this->_real()->{$name};
+    }
+
+    public function __set(string $name, mixed $value): void
+    {
+        $this->_real()->{$name} = $value;
+    }
+
+    public function __unset(string $name): void
+    {
+        unset($this->_real()->{$name});
+    }
+
+    public function __isset(string $name): bool
+    {
+        return isset($this->_real()->{$name});
+    }
+
+    public function __toString(): string
+    {
+        $object = $this->_real();
```

Comment:
At first, I emitted a deprecation here, saying 
> Class "Proxy" will not extend \Stringable anymore in 2.0. You should not use Proxy::__toString() nor cast a proxy to string.'

But it can cause some troubles because doctrine sometimes [casts the entity to a string](https://github.com/doctrine/orm/blob/2.16.x/lib/Doctrine/ORM/ORMInvalidArgumentException.php#L271) when it throws exceptions

I think we should not emit a deprecation here, because whereas we have old or new proxy system, the cast ends up calling real object's `__toString()`